src/utils/DataProcessingUtils.js

```diff
@@ -230,22 +230,14 @@ function processEntityValue(key :string, value :any, mappers :Map | Object) :any
   }
 
   if (isDefined(processedValue)) {
-    if (isValidDataPrimitive(processedValue)) {
-      return [processedValue];
-    }
-    if (isValidDataPrimitiveArray(processedValue)) {
+    if (Array.isArray(processedValue) || List.isList(processedValue)) {
       return processedValue;
     }
-    if (List.isList(processedValue)) {
-      LOG.error('entity values as immutable lists are not supported', processedValue);
-    }
-    if (Map.isMap(processedValue)) {
-      LOG.error('entity values as immutable maps are not supported', processedValue);
-    }
-    LOG.warn('processEntityValue() - unable to process value', processedValue);
```

Comment:
I think we should leave these in as they might be helpful with debugging during development.

Reply:
I removed them because the logged messages are incorrect. We are supporting immutable lists and maps in that they are acceptable. `processEntityValue` is value agnostic.